spec.md

```diff
@@ -49,6 +49,135 @@ The following will not be part of the specification:
+     them to the same consumer context.
+
+   In some cases, the consuming application might be interested in:
+   - obtaining further details about the event's subject from the
+     originating context, like obtaining detail information about a
+     changed object that requires privileged access authorization.
+     For example, a HR solution might only publish very limited
+     information in events for privacy reasons, and any event consumer
+     needing more data will have to obtain details related to the event
+     from the HR system under their own authorization context.
+   - interact with the event's subject at the originating context,
+     for instance reading a storage blob after having been informed
+     that this blob has just been created.
+
+   Consumer interests motivate requirements for which information
+   producers ought to include an event.
+
+3) Middleware routes events from producers to consumers, or onwards
+   to other middleware. Applications producing events might delegate
+   certain tasks arising from their consumers' requirements to
+   middleware:
+
+   - Management of many concurrent interested consumers for one of
+     multiple classes or originating contexts of events
+   - Processing of filter conditions over a class or originating context
+     of events on behalf of consumers.
+   - Transcoding, like encoding in MsgPack after decoding from JSON
+   - Transformation that changes the event's structure, like mapping from
+     a proprietary format to Cloud Events, while preserving the
+     identity and semantic integrity of the event.
+   - Instant "push-style" delivery to interested consumers.
+   - Storing events for eventual delivery, either for pick-up initiated
+     by the consumer ("pull"), or initiated by the middleware ("push")
+     after a delay.
+   - Observing event content or event flow for monitoring or
+     diagnostics purposes.
+
+   To satisfy these needs, middleware will be interested in:
+   - A metadata discriminator usable for classification or
+     contextualization of events so that consumers can express interest
+     in one or multiple such classes or contexts.
+     For instance, a consumer might be interested in all events related
+     to a specific directory inside a file storage account.
+   - A metadata discriminator that allows distinguishing the subject of
+     a particular event of that class or context.
+     For instance, a consumer might want to filter out all events related
+     to new files ending with ".jpg" (the file name being the "new file"
+     event's subject) for the context describing specific directory
+     inside a file storage account that it has registered interest on.
+   - An indicator for the encoding of the event and its data.
+   - An indicator for the structural layout (schema) for the event and
+     its data.
+
+   Whether its events are available for consumption via a middleware is
+   a delegation choice of the producer.
+
```

Comment:
Proposed addition based on conversation during March 23 call: "In practice, Middleware can take on role of a Producer when it changes the semantic meaning of an event, Consumers when it takes action based on an event, or Middleware when it routes events without making semantic changes." cc @duglin

Reply:
Like these additions!